src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -3,18 +3,28 @@
             [puppetlabs.services.protocols.request-handler :as handler]
             [puppetlabs.services.request-handler.request-handler-core :as core]
             [puppetlabs.services.jruby.jruby-puppet-service :as jruby]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [clojure.tools.logging :as log]))
 
 (defn- handle-request
-  [request jruby-service config]
-  (jruby/with-jruby-puppet jruby-puppet jruby-service
-    (core/handle-request request jruby-puppet config)))
+  [request jruby-service config puppet-version]
+  (try
+    (jruby/with-jruby-puppet jruby-puppet jruby-service
+                             (core/handle-request request jruby-puppet config))
+    (catch IllegalStateException e
```

Comment:
Not sure where this landed and I wouldn't block the PR on it or anything, but in general, I'm in favor of keeping the code in the `service` namespace to an absolute minimum... it should mostly just be the interactions with TK to get references to services / contexts / etc., and everything else should be pushed down into core if possible.